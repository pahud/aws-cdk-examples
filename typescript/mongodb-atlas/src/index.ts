export * from './bootstrap';
export * from './cluster';
export * from './database-user';
export * from './project';
export * from './project-ip-access-list';
export * from './types';
export * from './atlas-cluster';
export * from './serverless-instance';