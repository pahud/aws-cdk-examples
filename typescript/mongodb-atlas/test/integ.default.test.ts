import {
  App, Stack,
  aws_ec2 as ec2,
} from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { AtlasCluster, ServerlessInstance, ReplicationSpecs, EbsVolumeType, InstanceSize, AwsRegion, ClusterType } from '../src';
import { MongoDBAtlasBootstrap } from '../src/bootstrap';

test('default validation', () => {
  const app = new App();
  const bootstrapStack = new Stack(app, 'mongo-cdk-bootstrap');
  const demoStack = new Stack(app, 'mongodb-demo-stack');


  const secretProfile = 'my-mongo-profile';
  // bootstrap by creating the cfn extension execution role and profile secret.
  new MongoDBAtlasBootstrap(bootstrapStack, 'mongoCdkBootstrap', {
    roleName: 'cfn-ext-exec-role-for-mongo',
    secretProfile,
  });

  const replication: ReplicationSpecs[] = [
    {
      numShards: 1,
      advancedRegionConfigs: [
        {
          analyticsSpecs: {
            ebsVolumeType: EbsVolumeType.STANDARD,
            instanceSize: InstanceSize.M10,
            nodeCount: 1,
          },
          electableSpecs: {
            ebsVolumeType: EbsVolumeType.STANDARD,
            instanceSize: InstanceSize.M10,
            nodeCount: 3,
          },
          priority: 7,
          regionName: AwsRegion.US_EAST_1,
        },
      ],
    },
  ];

  const vpc = new ec2.Vpc(demoStack, 'Vpc', { natGateways: 1 });
  const orgId = process.env.MONGODB_ATLAS_ORG_ID || 'mock_id';

  // create a ReplicaSet cluster
  const cluster = new AtlasCluster(demoStack, 'Cluster', {
    clusterName: 'my-cluster',
    orgId,
    profile: secretProfile,
    replication,
    accessList: [{ ipAddress: vpc.vpcCidrBlock, comment: 'allow from my VPC only' }],
    peering: { vpc, cidr: '192.168.248.0/21' },
    clusterType: ClusterType.REPLICASET,
  });

  // create a serverless instance
  new ServerlessInstance(demoStack, 'ServerlessInstance', {
    instanceName: 'my-serverless-instance',
    profile: secretProfile,
    project: cluster.project,
    continuousBackup: true,
  });

  [bootstrapStack, demoStack].forEach(stack => {
    const t = Template.fromStack(stack);
    // should match snapshot
    expect(t).toMatchSnapshot();
  });
});