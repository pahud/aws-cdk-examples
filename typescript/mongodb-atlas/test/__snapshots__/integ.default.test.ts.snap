// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default validation 1`] = `
{
  "Outputs": {
    "mongoCdkBootstrapActivateCmdClusterC0C2877F": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws cloudformation activate-type --type-name MongoDB::Atlas::Cluster --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 --type RESOURCE --execution-role-arn ",
            {
              "Fn::GetAtt": [
                "mongoCdkBootstrapCfnExecRole6DEA467E",
                "Arn",
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapActivateCmdDatabaseUser9D62210B": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws cloudformation activate-type --type-name MongoDB::Atlas::DatabaseUser --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 --type RESOURCE --execution-role-arn ",
            {
              "Fn::GetAtt": [
                "mongoCdkBootstrapCfnExecRole6DEA467E",
                "Arn",
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapActivateCmdNetworkContainerB08CFFFE": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws cloudformation activate-type --type-name MongoDB::Atlas::NetworkContainer --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 --type RESOURCE --execution-role-arn ",
            {
              "Fn::GetAtt": [
                "mongoCdkBootstrapCfnExecRole6DEA467E",
                "Arn",
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapActivateCmdNetworkPeeringFF8C79D7": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws cloudformation activate-type --type-name MongoDB::Atlas::NetworkPeering --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 --type RESOURCE --execution-role-arn ",
            {
              "Fn::GetAtt": [
                "mongoCdkBootstrapCfnExecRole6DEA467E",
                "Arn",
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapActivateCmdProjectD99573BE": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws cloudformation activate-type --type-name MongoDB::Atlas::Project --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 --type RESOURCE --execution-role-arn ",
            {
              "Fn::GetAtt": [
                "mongoCdkBootstrapCfnExecRole6DEA467E",
                "Arn",
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapActivateCmdProjectIpAccessList3C42A46D": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws cloudformation activate-type --type-name MongoDB::Atlas::ProjectIpAccessList --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 --type RESOURCE --execution-role-arn ",
            {
              "Fn::GetAtt": [
                "mongoCdkBootstrapCfnExecRole6DEA467E",
                "Arn",
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapActivateCmdServerlessInstance2CA20B2E": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws cloudformation activate-type --type-name MongoDB::Atlas::ServerlessInstance --publisher-id bb989456c78c398a858fef18f2ca1bfc1fbba082 --type RESOURCE --execution-role-arn ",
            {
              "Fn::GetAtt": [
                "mongoCdkBootstrapCfnExecRole6DEA467E",
                "Arn",
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapMongoSecretProfileSecretNameC7B5BCEE": {
      "Value": {
        "Fn::Join": [
          "-",
          [
            {
              "Fn::Select": [
                0,
                {
                  "Fn::Split": [
                    "-",
                    {
                      "Fn::Select": [
                        6,
                        {
                          "Fn::Split": [
                            ":",
                            {
                              "Ref": "mongoCdkBootstrapMongoSecretProfilemymongoprofile24B3A06F",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
            {
              "Fn::Select": [
                1,
                {
                  "Fn::Split": [
                    "-",
                    {
                      "Fn::Select": [
                        6,
                        {
                          "Fn::Split": [
                            ":",
                            {
                              "Ref": "mongoCdkBootstrapMongoSecretProfilemymongoprofile24B3A06F",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
            {
              "Fn::Select": [
                2,
                {
                  "Fn::Split": [
                    "-",
                    {
                      "Fn::Select": [
                        6,
                        {
                          "Fn::Split": [
                            ":",
                            {
                              "Ref": "mongoCdkBootstrapMongoSecretProfilemymongoprofile24B3A06F",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
              ],
            },
          ],
        ],
      },
    },
    "mongoCdkBootstrapMongoSecretProfileUpdateSecretCommand31F9E864": {
      "Value": {
        "Fn::Join": [
          "",
          [
            "aws secretsmanager update-secret --secret-id ",
            {
              "Fn::Join": [
                "-",
                [
                  {
                    "Fn::Select": [
                      0,
                      {
                        "Fn::Split": [
                          "-",
                          {
                            "Fn::Select": [
                              6,
                              {
                                "Fn::Split": [
                                  ":",
                                  {
                                    "Ref": "mongoCdkBootstrapMongoSecretProfilemymongoprofile24B3A06F",
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                    ],
                  },
                  {
                    "Fn::Select": [
                      1,
                      {
                        "Fn::Split": [
                          "-",
                          {
                            "Fn::Select": [
                              6,
                              {
                                "Fn::Split": [
                                  ":",
                                  {
                                    "Ref": "mongoCdkBootstrapMongoSecretProfilemymongoprofile24B3A06F",
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                    ],
                  },
                  {
                    "Fn::Select": [
                      2,
                      {
                        "Fn::Split": [
                          "-",
                          {
                            "Fn::Select": [
                              6,
                              {
                                "Fn::Split": [
                                  ":",
                                  {
                                    "Ref": "mongoCdkBootstrapMongoSecretProfilemymongoprofile24B3A06F",
                                  },
                                ],
                              },
                            ],
                          },
                        ],
                      },
                    ],
                  },
                ],
              ],
            },
            " --secret-string "{\\"PublicKey\\":\\"\${MONGO_ATLAS_PUBLIC_KEY}\\",\\"PrivateKey\\":\\"\${MONGO_ATLAS_PRIVATE_KEY}\\"}"",
          ],
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "mongoCdkBootstrapCfnExecRole6DEA467E": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "resources.cloudformation.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AdministratorAccess",
              ],
            ],
          },
        ],
        "RoleName": "cfn-ext-exec-role-for-mongo",
      },
      "Type": "AWS::IAM::Role",
    },
    "mongoCdkBootstrapMongoSecretProfilemymongoprofile24B3A06F": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "Name": "cfn/atlas/profile/my-mongo-profile",
        "SecretString": "{"PublicKey":"changeMe", "PrivateKey": "changeMe"}",
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;

exports[`default validation 2`] = `
{
  "Outputs": {
    "ClusterVpcPeeringConnectionId734C7A5C": {
      "Value": {
        "Fn::GetAtt": [
          "ClusternetworkpeeringCluster3BFCD83E",
          "ConnectionId",
        ],
      },
    },
    "ClusterconnectionStrings17813228": {
      "Value": {
        "Fn::GetAtt": [
          "ClusterclusterCluster0934C304",
          "ConnectionStrings.StandardSrv",
        ],
      },
    },
    "ServerlessInstanceConnectionStringB32B56D6": {
      "Value": {
        "Fn::GetAtt": [
          "ServerlessInstance12909029",
          "ConnectionStrings.StandardSrv",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "ClusterCustomResourceVpcPeeringHandler93AA8AB3": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "ClusternetworkpeeringCluster3BFCD83E",
      ],
      "Properties": {
        "ConnectionId": {
          "Fn::GetAtt": [
            "ClusternetworkpeeringCluster3BFCD83E",
            "ConnectionId",
          ],
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "ClusterVpcPeeringProviderframeworkonEventC25E3ED3",
            "Arn",
          ],
        },
      },
      "Type": "Custom::VpcPeeringHandler",
      "UpdateReplacePolicy": "Delete",
    },
    "ClusterVpcPeeringHandler6D0A269A": {
      "DependsOn": [
        "ClusterVpcPeeringHandlerServiceRoleDefaultPolicyFA513A65",
        "ClusterVpcPeeringHandlerServiceRoleD02A4066",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "16913b75dfde1e173c826fe8ff63f11ae78620f808015d23d446080cc0b5e37a.zip",
        },
        "Handler": "vpc-peering-handler.on_event",
        "Role": {
          "Fn::GetAtt": [
            "ClusterVpcPeeringHandlerServiceRoleD02A4066",
            "Arn",
          ],
        },
        "Runtime": "python3.10",
      },
      "Type": "AWS::Lambda::Function",
    },
    "ClusterVpcPeeringHandlerServiceRoleD02A4066": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ClusterVpcPeeringHandlerServiceRoleDefaultPolicyFA513A65": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "ec2:AcceptVpcPeeringConnection",
                "ec2:DescribeVpcPeeringConnections",
                "ec2:DeleteVpcPeeringConnection",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":ec2:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":vpc-peering-connection/",
                      {
                        "Fn::GetAtt": [
                          "ClusternetworkpeeringCluster3BFCD83E",
                          "ConnectionId",
                        ],
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":ec2:",
                      {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":vpc/",
                      {
                        "Ref": "Vpc8378EB38",
                      },
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ClusterVpcPeeringHandlerServiceRoleDefaultPolicyFA513A65",
        "Roles": [
          {
            "Ref": "ClusterVpcPeeringHandlerServiceRoleD02A4066",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ClusterVpcPeeringProviderframeworkonEventC25E3ED3": {
      "DependsOn": [
        "ClusterVpcPeeringProviderframeworkonEventServiceRoleDefaultPolicy4A02D27A",
        "ClusterVpcPeeringProviderframeworkonEventServiceRole10788C15",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "8e3d635893ea17fa3158623489cd42c680fad925b38de1ef51cb10d84f6e245e.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (mongodb-demo-stack/Cluster/VpcPeeringProvider)",
        "Environment": {
          "Variables": {
            "USER_ON_EVENT_FUNCTION_ARN": {
              "Fn::GetAtt": [
                "ClusterVpcPeeringHandler6D0A269A",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": {
          "Fn::GetAtt": [
            "ClusterVpcPeeringProviderframeworkonEventServiceRole10788C15",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "ClusterVpcPeeringProviderframeworkonEventServiceRole10788C15": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ClusterVpcPeeringProviderframeworkonEventServiceRoleDefaultPolicy4A02D27A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "ClusterVpcPeeringHandler6D0A269A",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "ClusterVpcPeeringHandler6D0A269A",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ClusterVpcPeeringProviderframeworkonEventServiceRoleDefaultPolicy4A02D27A",
        "Roles": [
          {
            "Ref": "ClusterVpcPeeringProviderframeworkonEventServiceRole10788C15",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ClusterclusterCluster0934C304": {
      "DependsOn": [
        "ClusternetworkcontainerCluster7DC0968C",
      ],
      "Properties": {
        "ClusterType": "REPLICASET",
        "Name": "my-cluster",
        "Profile": "my-mongo-profile",
        "ProjectId": {
          "Fn::GetAtt": [
            "ClusterprojectCluster4966D242",
            "Id",
          ],
        },
        "ReplicationSpecs": [
          {
            "AdvancedRegionConfigs": [
              {
                "AnalyticsSpecs": {
                  "EbsVolumeType": "STANDARD",
                  "InstanceSize": "M10",
                  "NodeCount": 1,
                },
                "ElectableSpecs": {
                  "EbsVolumeType": "STANDARD",
                  "InstanceSize": "M10",
                  "NodeCount": 3,
                },
                "Priority": 7,
                "RegionName": "US_EAST_1",
              },
            ],
            "NumShards": 1,
          },
        ],
      },
      "Type": "MongoDB::Atlas::Cluster",
    },
    "ClusterdbuserClusterE136AF44": {
      "Properties": {
        "DatabaseName": "admin",
        "Password": "atlas-pwd",
        "Profile": "my-mongo-profile",
        "ProjectId": {
          "Fn::GetAtt": [
            "ClusterprojectCluster4966D242",
            "Id",
          ],
        },
        "Roles": [
          {
            "DatabaseName": "admin",
            "RoleName": "atlasAdmin",
          },
        ],
        "Username": "atlas-user",
      },
      "Type": "MongoDB::Atlas::DatabaseUser",
    },
    "ClusteripaccesslistCluster179AC96A": {
      "Properties": {
        "AccessList": [
          {
            "Comment": "allow from my VPC only",
            "IPAddress": {
              "Fn::GetAtt": [
                "Vpc8378EB38",
                "CidrBlock",
              ],
            },
          },
        ],
        "Profile": "my-mongo-profile",
        "ProjectId": {
          "Fn::GetAtt": [
            "ClusterprojectCluster4966D242",
            "Id",
          ],
        },
      },
      "Type": "MongoDB::Atlas::ProjectIpAccessList",
    },
    "ClusternetworkcontainerCluster7DC0968C": {
      "Properties": {
        "AtlasCidrBlock": "192.168.248.0/21",
        "Profile": "my-mongo-profile",
        "ProjectId": {
          "Fn::GetAtt": [
            "ClusterprojectCluster4966D242",
            "Id",
          ],
        },
        "Provisioned": true,
        "RegionName": "US_EAST_1",
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "MongoDB::Atlas::NetworkContainer",
    },
    "ClusternetworkpeeringCluster3BFCD83E": {
      "Properties": {
        "AccepterRegionName": {
          "Ref": "AWS::Region",
        },
        "AwsAccountId": {
          "Ref": "AWS::AccountId",
        },
        "ContainerId": {
          "Fn::GetAtt": [
            "ClusternetworkcontainerCluster7DC0968C",
            "Id",
          ],
        },
        "Profile": "my-mongo-profile",
        "ProjectId": {
          "Fn::GetAtt": [
            "ClusterprojectCluster4966D242",
            "Id",
          ],
        },
        "RouteTableCIDRBlock": {
          "Fn::GetAtt": [
            "Vpc8378EB38",
            "CidrBlock",
          ],
        },
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "MongoDB::Atlas::NetworkPeering",
    },
    "ClusterprojectCluster4966D242": {
      "Properties": {
        "Name": "project-projectCluster",
        "OrgId": "618cb365c835351df903f6ba",
        "Profile": "my-mongo-profile",
      },
      "Type": "MongoDB::Atlas::Project",
    },
    "ServerlessInstance12909029": {
      "Properties": {
        "ContinuousBackupEnabled": true,
        "Name": "my-serverless-instance",
        "Profile": "my-mongo-profile",
        "ProjectID": {
          "Fn::GetAtt": [
            "ClusterprojectCluster4966D242",
            "Id",
          ],
        },
        "ProviderSettings": {
          "ProviderName": "SERVERLESS",
          "RegionName": "US_EAST_1",
        },
        "TerminationProtectionEnabled": false,
      },
      "Type": "MongoDB::Atlas::ServerlessInstance",
    },
    "Vpc8378EB38": {
      "Properties": {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "VpcIGWD7BA715C": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "VpcPrivateSubnet1DefaultRouteBE02A9ED": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "VpcPublicSubnet1NATGateway4D7517AA",
        },
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet1RouteTableAssociation70C59FA6": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet1RouteTableB2C5B500",
        },
        "SubnetId": {
          "Ref": "VpcPrivateSubnet1Subnet536B997A",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet1RouteTableB2C5B500": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet1Subnet536B997A": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPrivateSubnet2DefaultRoute060D2087": {
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": {
          "Ref": "VpcPublicSubnet1NATGateway4D7517AA",
        },
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPrivateSubnet2RouteTableA678073B": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPrivateSubnet2RouteTableAssociationA89CAD56": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPrivateSubnet2RouteTableA678073B",
        },
        "SubnetId": {
          "Ref": "VpcPrivateSubnet2Subnet3788AAA1",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPrivateSubnet2Subnet3788AAA1": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet1DefaultRoute3DA9E72A": {
      "DependsOn": [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet1EIPD7E02669": {
      "Properties": {
        "Domain": "vpc",
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VpcPublicSubnet1NATGateway4D7517AA": {
      "DependsOn": [
        "VpcPublicSubnet1DefaultRoute3DA9E72A",
        "VpcPublicSubnet1RouteTableAssociation97140677",
      ],
      "Properties": {
        "AllocationId": {
          "Fn::GetAtt": [
            "VpcPublicSubnet1EIPD7E02669",
            "AllocationId",
          ],
        },
        "SubnetId": {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VpcPublicSubnet1RouteTable6C95E38E": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet1RouteTableAssociation97140677": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPublicSubnet1RouteTable6C95E38E",
        },
        "SubnetId": {
          "Ref": "VpcPublicSubnet1Subnet5C2D37C4",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet1Subnet5C2D37C4": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            0,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcPublicSubnet2DefaultRoute97F91067": {
      "DependsOn": [
        "VpcVPCGWBF912B6E",
      ],
      "Properties": {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": {
          "Ref": "VpcIGWD7BA715C",
        },
        "RouteTableId": {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VpcPublicSubnet2RouteTable94F7E489": {
      "Properties": {
        "Tags": [
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VpcPublicSubnet2RouteTableAssociationDD5762D8": {
      "Properties": {
        "RouteTableId": {
          "Ref": "VpcPublicSubnet2RouteTable94F7E489",
        },
        "SubnetId": {
          "Ref": "VpcPublicSubnet2Subnet691E08A3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VpcPublicSubnet2Subnet691E08A3": {
      "Properties": {
        "AvailabilityZone": {
          "Fn::Select": [
            1,
            {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": [
          {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          {
            "Key": "Name",
            "Value": "mongodb-demo-stack/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VpcVPCGWBF912B6E": {
      "Properties": {
        "InternetGatewayId": {
          "Ref": "VpcIGWD7BA715C",
        },
        "VpcId": {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
